### ISSUE TITLE
Inference results in extremely well structured noise even at 0.02 loss?

### ISSUE THREAD

**Issue by vedantroy** [lucidrains/imagen-pytorch#123]
Hi! I've run into a perplexing scenario. I'm so close to generating my first real image with this model.
I have a model trained for 90K+ steps, loss is 0.02. The model is trained on a single image.
The caption is just "sample_caption".

When I generate an image using the model, I get the following output:
![img_94114_0](https://user-images.githubusercontent.com/8407819/179291779-a027b2dd-e081-4daf-8640-d81c98bbb9ec.png)

It looks like extremely well-structured noise. But it's nothing like the original image.
I'm trying to debug what's going on here.

My imagen trainer is setup with the option `auto_normalize_img=True`, and my input to the trainer's forward function is a tensor with `dtype=torchuint8`.

Saving the sole training image to disk:
```
           # somewhere inside the training loop
            img = batch["img"]
            pil = T.to_pil_image(img.cpu()[0], mode="RGB")
            pil.save(f"./samples/img_truth.png", "PNG")
```

results in the original training image. So, I think the model is actually being trained on the correct inputs.

However, doing inference with the trained model with the following code:

```
    sample = "sample_caption"
    imgs = trainer.sample(texts=[sample], cond_scale=1., stop_at_unet_number=1)
    img = imgs[0]
   pil = T.to_pil_image(img.cpu(), mode="RGB")
   pil.save(f"./samples/img_{step}_{i}.png", "PNG")
```

just results in the aforementioned, extremely well-structured noise.
I was wondering if you might have an idea of what's happening here?
Maybe I'm not supposed to pass byte tensors to the trainer?

I'm using version: 0.22.9

**Comment by lucidrains** [comment]
@vedantroy yea, i think the problem was that you were passing in the image in as 0 to 255 uint8, but it actually expects 0 to 1 floats

**Comment by lucidrains** [comment]
@vedantroy however, in the most recent version, i auto convert the uint8 if i detect that type, so you should try again!

**Comment by vedantroy** [comment]
Quick question: When you do the auto-conversion you divide by 127.5. Is there a reason you don't divide by 255?
Mostly, I'm about to restart training, and I want to know if I should be dividing my byte tensor by 255 or 127.5 before passing it into the model (given that `auto_normalize_imgs=True`).

**Comment by lucidrains** [comment]
@vedantroy haha, yes i just fixed that sorry!
